### PR TITLE
Development

### PR DESCRIPTION
**Pull request by ahmadamira**
added the main and left section

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,23 +1,15 @@
-import logo from './logo.svg';
-import './App.css';
+import logo from "./logo.svg";
+import "./App.css";
```
Comment (omardere) [inline src/App.js]: remove unneeded imports

File: src/Components/Left-sidebar/Leftsidebar.css [hunk 2]
```diff
@@ -0,0 +1,48 @@
+.leftContainer {
+  width: 20%;
+  height: 1370px;
+  position: relative;
```
Comment (omardere) [inline src/Components/Left-sidebar/Leftsidebar.css:4]: also I think this code maybe make your UI not responsive

File: src/Components/Left-sidebar/Leftsidebar.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import "./Leftsidebar.css";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import Optionsbar from "./Optionsbar/Optionsbar";
+const Leftsidebar = () => {
```
Comment (omardere) [inline src/Components/Left-sidebar/Leftsidebar.js]: add space here to make your code more readable

File: src/Components/Left-sidebar/Leftsidebar.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+  return (
```
Comment (omardere) [inline src/Components/Left-sidebar/Leftsidebar.js:7]: space

File: src/Components/Left-sidebar/Leftsidebar.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
```
Comment (omardere) [inline src/Components/Left-sidebar/Leftsidebar.js:1]: pls use the name convention `LeftSidebar`

File: src/Components/Left-sidebar/Leftsidebar.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import Optionsbar from "./Optionsbar/Optionsbar";
+const Leftsidebar = () => {
+  return (
+    <div className="leftContainer">
+      <img src="/imgs/Vector.png" alt="My Logo" className="logo" />
+      <Optionsbar />
+      <button className="postButton">Post</button>
+      <div className="chatBox">
+        <img src="/imgs/myimg.jpg" alt="My Logo" className="profilelogo" />
```
Comment (omardere) [inline src/Components/Left-sidebar/Leftsidebar.js]: className="profileLogo" />

File: src/Components/Left-sidebar/Optionsbar/Optionsbar.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+import React from "react";
```
Comment (omardere) [inline src/Components/Left-sidebar/Optionsbar/Optionsbar.js:1]: pls use the name convention `OptionBar`

File: src/Components/Left-sidebar/Optionsbar/Optionsbar.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+import SearchIcon from "@mui/icons-material/Search";
+import NotificationsNoneIcon from "@mui/icons-material/NotificationsNone";
+import MailOutlineIcon from "@mui/icons-material/MailOutline";
+import ListAltIcon from "@mui/icons-material/ListAlt";
+import BookmarkBorderIcon from "@mui/icons-material/BookmarkBorder";
+import PeopleIcon from "@mui/icons-material/People";
+import PersonIcon from "@mui/icons-material/Person";
+import MoreIcon from "@mui/icons-material/More";
+import "./Optionbar.css";
+const optionsData = [
```
Comment (omardere) [inline src/Components/Left-sidebar/Optionsbar/Optionsbar.js:14]: space

File: src/Components/Left-sidebar/Optionsbar/Optionsbar.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    name: "Profile",
+  },
+  {
+    icon: <MoreIcon sx={{ color: "rgb(226, 223, 223)", height: "auto" }} />,
+    name: "More",
+  },
+];
+const Optionsbar = () => {
+  return (
+    <div className="optionbarContainer">
```
Comment (omardere) [inline src/Components/Left-sidebar/Optionsbar/Optionsbar.js]: optionBarContainer

File: src/Components/Mainsection/Followbutton/Followbutton.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import "./Followbutton.css";
+const Followbutton = () => {
```
Comment (omardere) [inline src/Components/Mainsection/Followbutton/Followbutton.js:4]: FollowButton

Comment (omardere) [inline src/Components/Mainsection/Followbutton/Followbutton.js:4]: space

File: src/Components/Mainsection/Graytext/Graytext.js [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import "./Graytext.css";
+const Graytext = ({ text }) => {
```
Comment (omardere) [inline src/Components/Mainsection/Graytext/Graytext.js:4]: space

File: src/Components/Mainsection/Headingsubheading/Headingsubheading.css [hunk 12]
```diff
@@ -0,0 +1,10 @@
+.headingcontainer {
```
Comment (omardere) [inline src/Components/Mainsection/Headingsubheading/Headingsubheading.css]: headingContainer

File: src/Components/Mainsection/Headingsubheading/Headingsubheading.js [hunk 13]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import "./Headingsubheading.css";
+const Headingsubheading = ({ heading, sub }) => {
```
Comment (omardere) [inline src/Components/Mainsection/Headingsubheading/Headingsubheading.js]: HeadingSubHeading
and I think the name should enough to be subHeading

Comment (omardere) [inline src/Components/Mainsection/Headingsubheading/Headingsubheading.js]: space

File: src/Components/Mainsection/Mainsection.css [hunk 14]
```diff
@@ -0,0 +1,103 @@
+.mainsectionContainer {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: mainSectionContainer

File: src/Components/Mainsection/Mainsection.css [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+  width: 50%;
+  position: relative;
+  border-right: 1px solid rgb(95, 94, 94);
+}
+.header {
+  display: flex;
+  flex-direction: row;
+}
+.reactimg {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: reactImage

File: src/Components/Mainsection/Mainsection.css [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+  width: 100%;
+}
+.belowHero {
+  color: white;
+  margin-left: 20px;
+}
+.linkanddate {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: linkAndDate

File: src/Components/Mainsection/Mainsection.css [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+  display: flex;
+  margin-bottom: 20px;
+}
+.reactlink {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: reactLink

File: src/Components/Mainsection/Mainsection.css [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+  color: blue;
+}
+.followers {
+  margin-left: 20px;
+}
+.navication {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: navigation

File: src/Components/Mainsection/Mainsection.css [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+.navButton {
+  border: none;
+  background: none;
+  padding: 0;
+  cursor: pointer;
+  margin-left: 10px;
+  color: rgb(95, 94, 94);
+  font-size: medium;
+}
+.navButtonicon {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: navButtonIcon

File: src/Components/Mainsection/Mainsection.css [hunk 20]
```diff
@@ -0,0 +1,103 @@
...
+  background: none;
+  padding: 2px;
+  cursor: pointer;
+  margin-left: 10px;
+  color: rgb(95, 94, 94);
+  font-size: medium;
+  border: 1px solid rgb(95, 94, 94);
+  border-radius: 50%;
+}
+.navButtonfollow {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: navButtonFollow

File: src/Components/Mainsection/Mainsection.css [hunk 21]
```diff
@@ -0,0 +1,103 @@
...
+  font-size: medium;
+  border: 1px solid rgb(95, 94, 94);
+  border-radius: 20px;
+}
+.btnDiv {
+  margin-right: 20px;
+  margin-top: 20px;
+  text-align: right;
+}
+.showmorebtn {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: showMoreBtn

File: src/Components/Mainsection/Mainsection.css [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+  margin-left: 10px;
+  color: white;
+  font-size: medium;
+  border-bottom: 4px solid rgb(95, 192, 252);
+}
+hr {
+  border: none;
+  border-top: 2px solid rgb(95, 94, 94);
+}
+.prophilephoto {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.css]: profilePhoto

File: src/Components/Mainsection/Mainsection.js [hunk 23]
```diff
@@ -0,0 +1,106 @@
...
+import Headingsubheading from "./Headingsubheading/Headingsubheading";
+import InsertLinkIcon from "@mui/icons-material/InsertLink";
+import CalendarMonthIcon from "@mui/icons-material/CalendarMonth";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import NotificationAddIcon from "@mui/icons-material/NotificationAdd";
+import "./Mainsection.css";
+import Graytext from "./Graytext/Graytext";
+import Post from "./Post/Post";
+import Whotofollow from "./Whotofollow/Whotofollow";
+const data = [
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.js:13]: space

File: src/Components/Mainsection/Mainsection.js [hunk 24]
```diff
@@ -0,0 +1,106 @@
...
+  {
+    img: "/imgs/freecodecamp.jpg",
+    name: "freeCodeCamp.org ",
+    subname: "@freeCodeCamp",
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.js:17]: subName

File: src/Components/Mainsection/Mainsection.js [hunk 25]
```diff
@@ -0,0 +1,106 @@
...
+  },
+  {
+    img: "/imgs/TailWindCSS.png",
+    name: "TailWind CSS",
+    subname: "@TailWindCSS",
+    followers: "AhmadAmira and 3 others follow",
+    text: "The utility-first CSS framework. Rapidly build modern websites, without ever leaving your HTML.",
+  },
+];
+const Mainsection = () => {
```
Comment (omardere) [inline src/Components/Mainsection/Mainsection.js]: space

Comment (omardere) [inline src/Components/Mainsection/Mainsection.js]: MainSection

File: src/Components/Mainsection/Post/Post.css [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+}
+.postContent {
+  margin-top: 30px;
+  margin: 0px;
+  padding: 0px;
+}
+.flexdiv {
+  display: flex;
+}
+.postOwnerphoto {
```
Comment (omardere) [inline src/Components/Mainsection/Post/Post.css]: postOwnerPhoto

File: src/Components/Mainsection/Post/Post.css [hunk 27]
```diff
@@ -0,0 +1,44 @@
...
+  color: white;
+  margin-top: 30px;
+  justify-content: space-between;
+}
+.postContent {
+  margin-top: 30px;
+  margin: 0px;
+  padding: 0px;
+}
+.flexdiv {
```
Comment (omardere) [inline src/Components/Mainsection/Post/Post.css]: pls use  more appropriate name

File: src/Components/Mainsection/Post/Post.css [hunk 28]
```diff
@@ -0,0 +1,44 @@
...
+  margin-bottom: 20px;
+  border-radius: 50%;
+}
+.postName {
+  margin-left: 10px;
+}
+.postOwner {
+  display: flex;
+}
+.moreDiv {
```
Comment (omardere) [inline src/Components/Mainsection/Post/Post.css]: pls use more appropriate name

File: src/Components/Mainsection/Post/Post.js [hunk 29]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import "./Post.css";
+import ModeCommentIcon from "@mui/icons-material/ModeComment";
+import SwapCallsIcon from "@mui/icons-material/SwapCalls";
+import FavoriteBorderIcon from "@mui/icons-material/FavoriteBorder";
+import DriveFolderUploadIcon from "@mui/icons-material/DriveFolderUpload";
+import Graytext from "../Graytext/Graytext";
+const Post = () => {
```
Comment (omardere) [inline src/Components/Mainsection/Post/Post.js:10]: space

File: src/Components/Mainsection/Whotofollow/Whotofollow.css [hunk 30]
```diff
@@ -0,0 +1,30 @@
+.whofollows {
```
Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.css]: whoFollows

File: src/Components/Mainsection/Whotofollow/Whotofollow.css [hunk 31]
```diff
@@ -0,0 +1,30 @@
...
+.followContainer {
+  margin: 20px;
+}
+.followBox {
+  display: flex;
+
+  width: 100%;
+  color: white;
+}
+.followlogo {
```
Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.css]: followLogo

File: src/Components/Mainsection/Whotofollow/Whotofollow.css [hunk 32]
```diff
@@ -0,0 +1,30 @@
...
+  width: 40px;
+  height: 40px;
+  display: block;
+  margin-top: 10px;
+  margin-left: 10px;
+  margin-right: 20px;
+  margin-bottom: 20px;
+  border-radius: 50%;
+}
+.headingfollow {
```
Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.css]: headingFollow

File: src/Components/Mainsection/Whotofollow/Whotofollow.js [hunk 33]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import "./Whotofollow.css";
+import PersonIcon from "@mui/icons-material/Person";
+import Graytext from "../Graytext/Graytext";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
```
Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.js]: pls remove unneeded import

File: src/Components/Mainsection/Whotofollow/Whotofollow.js [hunk 34]
```diff
@@ -0,0 +1,28 @@
...
+import Headingsubheading from "../Headingsubheading/Headingsubheading";
+import Followbutton from "../Followbutton/Followbutton";
+
+const Whotofollow = ({ img, name, subname, followers, text }) => {
```
Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.js]: WhoToFollow

Comment (omardere) [inline src/Components/Mainsection/Whotofollow/Whotofollow.js]: subName

File: src/Components/Rightsection/Rightsection.css [hunk 35]
```diff
@@ -0,0 +1,31 @@
+.rightsectioncontainer {
```
Comment (omardere) [inline src/Components/Rightsection/Rightsection.css]: rightSectionContainer

File: src/Components/Rightsection/Rightsection.css [hunk 36]
```diff
@@ -0,0 +1,31 @@
...
+  width: 30%;
+}
+/* SearchBar.css */
```
Comment (omardere) [inline src/Components/Rightsection/Rightsection.css]: remove

File: src/Components/Rightsection/Rightsection.css [hunk 37]
```diff
@@ -0,0 +1,31 @@
...
+.search-bar {
+  margin-top: 20px;
+  margin-left: 20px;
+  width: 80%;
+  height: 20px;
+  display: flex;
+  align-items: center;
+  background-color: rgb(68, 67, 67); /* Gray background color */
```
Comment (omardere) [inline src/Components/Rightsection/Rightsection.css]: remove the comment

File: src/Components/Rightsection/Rightsection.js [hunk 38]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import "./Rightsection.css";
+import SearchIcon from "@mui/icons-material/Search";
+import Youmightlike from "./Youmightlike/Youmightlike";
+import Trendsforyou from "./Trendsforyou/Trendsforyou";
+const Rightsection = () => {
```
Comment (omardere) [inline src/Components/Rightsection/Rightsection.js]: space

Comment (omardere) [inline src/Components/Rightsection/Rightsection.js]: RightSection

File: src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.css [hunk 39]
```diff
@@ -0,0 +1,8 @@
+.tfyheader {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.css]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.css [hunk 40]
```diff
@@ -0,0 +1,8 @@
...
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: 20px;
+}
+.Tfrcardcontainer {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.css]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.js [hunk 41]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import Graytext from "../../../Mainsection/Graytext/Graytext";
+import MoreHorizIcon from "@mui/icons-material/MoreHoriz";
+import Headingsubheading from "../../../Mainsection/Headingsubheading/Headingsubheading";
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.js]: remove unused import

File: src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.js [hunk 42]
```diff
@@ -0,0 +1,21 @@
...
+import "./Tfucard.css";
+const Tfucard = () => {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.js]: space

Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Tfycard/Tfucard.js]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Trendsforyou.css [hunk 43]
```diff
@@ -0,0 +1,14 @@
+.tfycontainer {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Trendsforyou.css]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Trendsforyou.css [hunk 44]
```diff
@@ -0,0 +1,14 @@
...
+  color: white;
+  margin-left: 20px;
+  width: 85%;
+  background-color: rgb(39, 39, 39);
+  border-radius: 20px;
+  margin-top: 20px;
+}
+.tfyheader {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Trendsforyou.css]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Trendsforyou.js [hunk 45]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import "./Trendsforyou.css";
+import Tfucard from "./Tfycard/Tfucard";
+import Graytext from "../../Mainsection/Graytext/Graytext";
+const data = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+const Trendsforyou = () => {
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Trendsforyou.js:8]: space

Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Trendsforyou.js:8]: pls use more appropriate name

File: src/Components/Rightsection/Trendsforyou/Trendsforyou.js [hunk 46]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div>
+      <div className="tfycontainer">
+        <h3 className="tfyheader">Trends for you</h3>
+        {data.map((item, index) => (
+          <Tfucard />
```
Comment (omardere) [inline src/Components/Rightsection/Trendsforyou/Trendsforyou.js]: use key and remove unused item
`data.map((_, index)`

File: src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.css [hunk 47]
```diff
@@ -0,0 +1,21 @@
+.ymlBox {
+  display: flex;
+
+  width: 100%;
+  color: white;
+}
+.ymllogo {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.css]: youMightLikeLogo

File: src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.css [hunk 48]
```diff
@@ -0,0 +1,21 @@
...
+  width: 40px;
+  height: 40px;
+  display: block;
+  margin-top: 10px;
+  margin-left: 10px;
+  margin-right: 20px;
+  margin-bottom: 20px;
+  border-radius: 50%;
+}
+.Ymlcardcontainer {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.css]: youMightLikeCard

File: src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.js [hunk 49]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import "./Ymlcard.css";
+import Headingsubheading from "../../../Mainsection/Headingsubheading/Headingsubheading";
+import Followbutton from "../../../Mainsection/Followbutton/Followbutton";
+const Ymlcard = ({ img, name, subname }) => {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.js]: space

Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Ymlcard/Ymlcard.js]: youMightLikeCard

File: src/Components/Rightsection/Youmightlike/Youmightlike.css [hunk 50]
```diff
@@ -0,0 +1,11 @@
+.ymlcontainer {
+  color: white;
+  margin-left: 20px;
+  width: 85%;
+  background-color: rgb(39, 39, 39);
+  border-radius: 20px;
+  margin-top: 20px;
+}
+.ymlheader {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Youmightlike.css]: youMightLikeHeader

File: src/Components/Rightsection/Youmightlike/Youmightlike.css [hunk 51]
```diff
@@ -0,0 +1,11 @@
+.ymlcontainer {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Youmightlike.css]: youMightLikeContainer

File: src/Components/Rightsection/Youmightlike/Youmightlike.js [hunk 52]
```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import "./Youmightlike.css";
+import Ymlcard from "./Ymlcard/Ymlcard";
+const data = [
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Youmightlike.js:4]: space

File: src/Components/Rightsection/Youmightlike/Youmightlike.js [hunk 53]
```diff
@@ -0,0 +1,38 @@
...
+  {
+    img: "/imgs/freecodecamp.jpg",
+    name: "freeCodeCamp.org ",
+    subname: "@freeCodeCamp",
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Youmightlike.js:8]: subName

File: src/Components/Rightsection/Youmightlike/Youmightlike.js [hunk 54]
```diff
@@ -0,0 +1,38 @@
...
+    name: "TypeScript",
+    subname: "@Typescript",
+  },
+  {
+    img: "/imgs/TailWindCSS.png",
+    name: "TailWind CSS",
+    subname: "@TailWindCSS",
+  },
+];
+const Youmightlike = () => {
```
Comment (omardere) [inline src/Components/Rightsection/Youmightlike/Youmightlike.js]: youMightLike

**Comment by ahmadamira** [comment]
I made the changes

**Pull request merged**